src/libpcp_web/src/query.c

```diff
@@ -2361,6 +2361,14 @@ series_expr_canonical(node_t *np, int idx)
     case N_SUM_SAMPLE:
     case N_STDEV_INST:
     case N_STDEV_SAMPLE:
+    case N_NTH_PERCENTILE_INST:
+    	left = series_expr_canonical(np->left, idx);
+	right = series_expr_canonical(np->right, idx);
+	break;
+    case N_NTH_PERCENTILE_SAMPLE:
+    	left = series_expr_canonical(np->left, idx);
+	right = series_expr_canonical(np->right, idx);
+	break;
```

Comment:
the above section can be compressed (follow the switch/case pattern there - we only need to do the left/right assignments once).